src/domain/shared/ContextManager.ts

```diff
@@ -0,0 +1,96 @@
+//
+//  ContextManager.ts
+//
+//  Created by David Rowe on 7 Aug 2021.
+//  Copyright 2021 Vircadia contributors.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+
+type DependencyType = {
+    name: string,
+    new(...args: any[]): object  // eslint-disable-line @typescript-eslint/no-explicit-any, @typescript-eslint/ban-types
+};
+
+type DependencyObject = {
+    //
+};
+
+// A context, mapping class names to class instances.
+type Context = Map<string, DependencyObject>;
+
+// All the contexts.
+type ContextTable = Array<Context>;
+
+
+/*@devdoc
+ *  The <code>ContextManager</code> namespace manages sets of unique object instances. Such a set forms a "context" in which
```

Comment:
What is an object instance? An object being... a DomainServer? I think the context manager and contexts in general might need more explaining as to what they are.

Reply:
I've updated the documentation.